llvm/lib/Target/AIE/AIEPostSelectOptimize.cpp

```diff
@@ -438,6 +440,140 @@ bool duplicateAdressingRegs(MachineBasicBlock &MBB, MachineRegisterInfo &MRI) {
   return tryToDuplicateLoadUse(LoadUses, NonLoadUses, MRI, TII);
 }
 
+using Operation = AIEBaseInstrInfo::AbstractVecOp::OperationType;
+using AbstractVecOp = AIEBaseInstrInfo::AbstractVecOp;
+using OptionalVecOp = std::optional<AbstractVecOp>;
+
+bool getGlobalValue(const MachineInstr *MI,
+                    SmallPtrSet<const Value *, 4> &GVSet,
+                    MachineRegisterInfo &MRI) {
+
+  MI = getDefIgnoringCopiesAndBitcasts(MI->getOperand(1).getReg(), MRI);
+
+  for (auto MO : MI->uses()) {
+    if (MO.isGlobal()) {
+      GVSet.insert(MO.getGlobal());
+      return true;
+    }
```

Comment:
MI could be anything, e.g. the top of a tree of selects. I guess we have to return false whenever MO isn't a global and collect all used globals.

Reply:
Yes I believe any operand that cannot be tracked as a memory location should make us return `false`

Reply:
Done! Now we consider only the case with one src operand that loads a global.